File: Php/core/FormFields/LocationFieldModel.php
namespace core\FormFields{
	use core\Globalization;

	class LocationFieldModel extends ListFieldModel{

		public function __construct($fieldName, $fieldId, $selectedValue, $locations, $firstItemText = null){
			$items = $this->getSelectItems($locations);
			parent::__construct($fieldName, $fieldId, $selectedValue, $items, $firstItemText);
		}

		private function getSelectItems($locations){
			$items = array();
			if($locations != null){
				foreach($locations as $location){
					$items[] = new SelectFieldItem($location);
				}
			}
			return $items;
		}

	}

}

File: Web/Models/FormModels/LocationFieldModel.cs
using System.Collections.Generic;
using System.Linq;

namespace Web.Models.FormModels{

	public class LocationFieldModel : ListFieldModel{

		public LocationFieldModel(string fieldName, string fieldId, string selectedValue, IEnumerable<string> locations, string firstItemText = null)
            : base(fieldName, fieldId, selectedValue, GetSelectItems(locations), firstItemText)
        {
		}

		private static IEnumerable<SelectFieldItem> GetSelectItems(IEnumerable<string> locations)
		{
		    var items = new List<SelectFieldItem>();
			if(locations != null){
			    items.AddRange(locations.Select(location => new SelectFieldItem(location)));
			}
			return items;
		}

	}

}